domainobjs/ts/__tests__/commands.test.ts

```diff
@@ -1,120 +1,160 @@
+      const c1: PCommand = new PCommand(BigInt(10), newPubKey, BigInt(0), BigInt(9), BigInt(1), BigInt(123));
 
-    // shallow copy
-    const c2 = c1;
-    c1.nonce = BigInt(9999);
-    expect(c1.nonce.toString()).to.eq(c2.nonce.toString());
+      // shallow copy
+      const c2 = c1;
+      c1.nonce = BigInt(9999);
+      expect(c1.nonce.toString()).to.eq(c2.nonce.toString());
 
-    // deep copy
-    const c3 = c1.copy();
-    c1.nonce = BigInt(8888);
+      // deep copy
+      const c3 = c1.copy();
+      c1.nonce = BigInt(8888);
 
-    expect(c1.nonce.toString()).not.to.eq(c3.nonce.toString());
-  });
+      expect(c1.nonce.toString()).not.to.eq(c3.nonce.toString());
+    });
 
-  it("message.copy() should produce a deep copy", () => {
-    const m1 = new Message(BigInt(1), [
-      BigInt(2),
-      BigInt(3),
-      BigInt(4),
-      BigInt(5),
-      BigInt(6),
-      BigInt(7),
-      BigInt(8),
-      BigInt(9),
-      BigInt(10),
-      BigInt(11),
-    ]);
-
-    const m2 = m1.copy();
-    expect(m2.equals(m1)).to.eq(true);
-  });
+    it("should produce a deep copy for TCommand", () => {
+      const c1: TCommand = new TCommand(BigInt(10), BigInt(0), BigInt(9));
 
-  it("message.asCircuitInputs() should return a array", () => {
-    const m1 = new Message(BigInt(1), [
-      BigInt(2),
-      BigInt(3),
-      BigInt(4),
-      BigInt(5),
-      BigInt(6),
-      BigInt(7),
-      BigInt(8),
-      BigInt(9),
-      BigInt(10),
-      BigInt(11),
-    ]);
-
-    const arr = m1.asCircuitInputs();
-    expect(arr.length).to.eq(11);
-    expect(arr[0]).to.eq(BigInt(1));
-    expect(arr[1]).to.eq(BigInt(2));
-    expect(arr[2]).to.eq(BigInt(3));
-    expect(arr[3]).to.eq(BigInt(4));
-    expect(arr[4]).to.eq(BigInt(5));
-    expect(arr[5]).to.eq(BigInt(6));
-    expect(arr[6]).to.eq(BigInt(7));
-    expect(arr[7]).to.eq(BigInt(8));
-    expect(arr[8]).to.eq(BigInt(9));
-    expect(arr[9]).to.eq(BigInt(10));
-    expect(arr[10]).to.eq(BigInt(11));
-    expect(arr).to.be.instanceOf(Array);
+      // shallow copy
+      const c2 = c1;
+      c1.amount = BigInt(9999);
+      expect(c1.amount.toString()).to.eq(c2.amount.toString());
+
+      // deep copy
+      const c3 = c1.copy();
+      c1.amount = BigInt(8888);
+
+      expect(c1.amount.toString()).not.to.eq(c3.amount.toString());
+    });
   });
 
-  describe("TCommand", () => {
-    const stateIndex = BigInt(0);
-    const amount = BigInt(100);
-    const pollId = BigInt(1);
-    const tCommand = new TCommand(stateIndex, amount, pollId);
+  describe("deserialization/serialization", () => {
+    describe("toJSON", () => {
+      it("should produce a JSON object with valid values", () => {
+        const c1: TCommand = new TCommand(BigInt(10), BigInt(0), BigInt(9));
+        const json = c1.toJSON();
+        expect(json).to.not.eq(null);
+        expect(json.stateIndex).to.eq("10");
+        expect(json.amount).to.eq("0");
+        expect(json.pollId).to.eq("9");
```

Comment:
should check the `json.cmdType` as well?

Reply:
aren't we already?

Reply:
Just because I saw we check the `expect(json.cmdType).to.eq("1")` in the `PCommand` part below, so I think it should be checked also for `TCommand`.

Reply:
Oh I see what you mean, my bad I missed the one above! Will add the check there too